### PR TITLE
Acolin for BTB source

### PR DESCRIPTION
**Pull request by MaxTousss**
Goal: enable simulation of acolin for back-to-back source

TODO:
- [x] Implement acolin (Currently using my cheap way, not the nice boost thingy)
- [x] Provide an interface to set the acolin value 
- [x] Add test(s) of the behavior (Tested by hand, going to add along side test079)
- [x] Clean up the code
  - [x] `Pi` should be a constant somewhere...
  - [x] There should be a way to let user use `deg` or `radian` units, right? For example, activity can be set with `* Bq` or `* MBq`
  - [x] ~Use the try/catch of setting acolin FWHM for verbose to tell the user that the default value is used~ Marked as todo in the code
- [x] Document how to use this feature
- [x] Document the existence of back_to_back source
- [x] ~Add hook for python to get the current value of acollinearity~ Assuming that initialization value is good enough 

@dsarrut I have a couple of questions:
- What is the difference between a back-to-back and gamma source? BTB is a gamma source where the energy is set to 511 kev?
- Since we are starting anew, should we rename back-to-back to something more adequate since we are adding acolin? e.g. `511kevPhotonPair`?
- Could you quickly look at the cleanup elements of the TODO? You might be able to guide me on where I can look to achieve them.

### CODE REVIEW

**Comment by dsarrut** [comment]
> * What is the difference between a back-to-back and gamma source? BTB is a gamma source where the energy is set to 511 kev?

Yes btb is a simple gamma source with fixed energy that create **two** primary gamma each event vertex:
see:  https://github.com/OpenGATE/opengate/blob/master/core/opengate_core/opengate_lib/GateSingleParticleSource.cpp#L122

> * Since we are starting anew, should we rename back-to-back to something more adequate since we are adding acolin? e.g. `511kevPhotonPair`?

Unsure, my feeling is that back to back is a common way to name this source, the acolin can be view as an option. However, we can ask other colleagues to know what they are thinking about it. 

> * Could you quickly look at the cleanup elements of the TODO? You might be able to guide me on where I can look to achieve them.
>   * [ ]  `Pi` should be a constant somewhere...
```
import math
print(math.pi)
```
or 
```
import numpy
print(numpy.pi)
```

>   * [ ]  There should be a way to let user use `deg` or `radian` units, right? For example, activity can be set with `* Bq` or `* MBq`

yes, same principle, this is geant4 units systems that we put in g4_units:
```
angle = 66*g4_units.deg
angle = numpy.pi/2*g4_units.rad
```
Then if you use a function that, say, need deg, and you let the user use what he wants, you do: 
```
my_function_that_need_deg(angle/g4_units.deg)
```

>   * [ ]  Use the try/catch of setting acolin FWHM for verbose to tell the user that the default value is used
Unsure, I think the default value will be in the user options (python side). From the user, I imagine something like :

```
my_b2b_source.use_acolinearity = True
my_b2b_source.accolinearity_fwhm = 666 * deg
```

thanks!

**Comment by MaxTousss** [comment]
When you say "we can ask colleagues", do you mean dev. or users? If it is the latter, I could send a poll in the mailing list or something like that. Or, you could ask at a future GATE meeting, in order to get a more involved discussion/vote.

As for `pi` and units, sorry I forgot to mention that it is for c++ part. I will add review comments to show where.

**Comment by MaxTousss** [comment]
Adding documentation is put on hold until #563 is resolved...

**Comment by dsarrut** [comment]
thanks @MaxTousss ! I will have a look in the coming days

**Comment by MaxTousss** [comment]
Documentation done. Only need to know what to do with the remaining questions

**Comment by MaxTousss** [comment]
Found how to deal with units.
Added a TODO about creating a hook, availble in python?, to get the current value of acollinearity.

**Comment by MaxTousss** [comment]
@dsarrut When you have the time, I would request some inputs on the remaining two TODOs. They are more quality of life things, but I do not know if it is required in GATE, nor how to do them atm.

**Comment by MaxTousss** [comment]
@dsarrut  Changed default of b2b aco to 0.5 deg (you had set it has 0.55 deg) since I used that value everywhere else. If you prefer 0.55 deg., I will change the code and tests (current custom aco is 0.5 deg.!) accordingly.

Also, since the default is set on the python side, the try/catch and default value in GateGenericSource are no longer relevant and potential point of failure, right? (Potential point of failure since the default values is set in two different places) However, you didn't remove them, is there a reason?

File: core/opengate_core/opengate_lib/GateSingleParticleSource.h [hunk 1]
```diff
@@ -56,6 +56,10 @@ class GateSingleParticleSource : public G4VPrimaryGenerator {
 
   void SetBackToBackMode(bool flag, bool accolinearityFlag);
 
+  // Probably an underestimation in most cases, but it is the most cited
+  // value (Moses 2011)
+  void SetAccolinearityFWHM(double accolinearityFWHM = 0.5 * CLHEP::deg);
```
[the remark this answers is not in the document]
Comment (dsarrut) [reply core/opengate_core/opengate_lib/GateSingleParticleSource.h:61]: yes, the default should be removed I think (cos already on the python side, easier to change)

**Pull request merged**